lib/tapioca/gem/plugins/yard_doc.rb

```diff
@@ -0,0 +1,62 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "pathname"
+
+module Tapioca
+  module Gem
+    module Plugins
+      class YardDoc < Base
+        extend T::Sig
+
+        IGNORED_COMMENTS = T.let([
+          ":doc:",
+          ":nodoc:",
+          "typed:",
+          "frozen_string_literal:",
+          "encoding:",
+          "warn_indent:",
+          "shareable_constant_value:",
+          "rubocop:",
+        ], T::Array[String])
+
+        sig { override.params(mod: RBI::Module).void }
+        def decorate_module(mod)
+          mod.comments = documentation_comments(mod.fully_qualified_name)
+        end
+
+        sig { override.params(cls: RBI::Class).void }
+        def decorate_class(cls)
+          cls.comments = documentation_comments(cls.fully_qualified_name)
+        end
+
+        sig { params(const: RBI::Const).void }
+        def decorate_const(const)
+          const.comments = documentation_comments(const.fully_qualified_name)
+        end
+
+        sig { override.params(meth: RBI::Method).void }
+        def decorate_method(meth)
+          meth.comments = documentation_comments(meth.fully_qualified_name)
+        end
```

Comment:
Looks like the only downside is having 4 of these nearly identical method calls on different `RBI` objects for each plugin. In more complicated plugins I assume we can have an "entrypoint" method that is responsible of the internal plugin logic, instead of `SymbolTableCompiler` owning driving it with methods like https://github.com/Shopify/tapioca/blob/53b7307ddb357845f5166a6d57161e5de6300547/lib/tapioca/compilers/symbol_table_compiler.rb#L195